### PR TITLE
Tip of the Week: Remove Unused Code to Avoid Software Decay

### PR DESCRIPTION
**Pull request by d33bs**
This PR adds a new tip of the week titled "Remove Unused Code to Avoid Software Decay". It depends on another PR for an example notebook which is linked from the content. 

Thank you for any comments, suggestions, or feedback you may have!

### CODE REVIEW

File: _posts/2022-12-12-Remove-Unused-Code-to-Avoid-Decay.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+unused_code_example.py:4: unused import 'pd' (90% confidence)
+unused_code_example.py:7: unused function 'unused_function' (60% confidence)
+unused_code_example.py:14: unused variable 'unused_var' (60% confidence)
+```
+
+_Example of Vulture command line usage to discover unused code._
+
+[Vulture](https://github.com/jendrikseipp/vulture) is one tool dedicated to finding unused python code. Vulture provides both a command line interface and Python API for discovering unused code. It also provide a rough confidence to show how certain it was about whether the block of code was unused. See the following interactive example for a demonstration of using Vulture.
+
+[Example Link](https://cu-dbmi.github.io/notebooks/lab?path=unused_code_detection.ipynb)
```
Comment (falquaddoomi) [inline _posts/2022-12-12-Remove-Unused-Code-to-Avoid-Decay.md:111]: I'd prefer to see this link labeled by what it points to, e.g., "Unused Code Detection Example".

File: _posts/2022-12-12-Remove-Unused-Code-to-Avoid-Decay.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+## Further Code Usefulness Detection with Pylint and Coverage.py
+
+In addition to Vulture, [Pylint](https://pylint.pycqa.org/en/latest/index.html) and [Coverage.py](https://coverage.readthedocs.io/) can be used in a similar way to help show where code may not have been used within your project.
+
+[Pylint](https://pylint.pycqa.org/en/latest/index.html) focuses on code style and other static analysis in addition to unused variables. See [Pylint's Checkers](https://pylint.pycqa.org/en/latest/user_guide/checkers/features.html) page for more details here, using "unused-*" as a reference to checks it performs which focus on unused code.
+
+[Coverage.py](https://coverage.readthedocs.io/) helps show you which parts of your code have been executed or not. A common usecase for Coverage involves measuring "test coverage", or which parts of your code are executed in relationship to tests written for that code. This provides another perspective on code utility; if there's not a test for the code, is it worth keeping?
```
Comment (falquaddoomi) [inline _posts/2022-12-12-Remove-Unused-Code-to-Avoid-Decay.md:119]: Extremely minor nitpick, but I prefer a colon over a semicolon if the clause following it is heavily implied, i.e.:
```suggestion
[Coverage.py](https://coverage.readthedocs.io/) helps show you which parts of your code have been executed or not. A common usecase for Coverage involves measuring "test coverage", or which parts of your code are executed in relationship to tests written for that code. This provides another perspective on code utility: if there's not a test for the code, is it worth keeping?
```

**Review by falquaddoomi — COMMENTED** [review]
Nice work! I had a few optional comments, but you could publish it as-is in my opinion. I like the use of diagrams, charts, and that little hiker to make it more visually appealing.

**Comment by seandavi** [comment]
Just adding my kudos! Looks fantastic!

File: _posts/2022-12-12-Remove-Unused-Code-to-Avoid-Decay.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  - vulture
+  - pylint
+  - coverage.py
+---
+
+# Tip of the Week: Remove Unused Code to Avoid Software Decay
+
+> Each week we seek to provide a software tip of the week geared towards helping you achieve your software goals. If you have any software questions or suggestions for an upcoming tip of the week, please don’t hesitate to reach out to #software-engineering on Slack or email DBMISoftwareEngineering at olucdenver.onmicrosoft.com
+
+The act of creating software often involves many iterations of writing, personal collaborations, and testing. During this process it's common to lose awareness of code which is no longer used (and may not be tested or otherwise linted). Unused code may contribute to ["software decay"](https://en.wikipedia.org/wiki/Software_rot) (the gradual diminishment of code quality or functionality). This post will cover software decay and strategies for addressing unused code to help keep your code quality high.
```
[the remark this answers is not in the document]
Comment (d33bs) [reply _posts/2022-12-12-Remove-Unused-Code-to-Avoid-Decay.md:20]: Thank you, agreed - this rephrasing is a lot more potent. I'll add this into the commits.

File: _posts/2022-12-12-Remove-Unused-Code-to-Avoid-Decay.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+__TLDR (too long, didn't read);__
+Unused code is easy to amass and may cause your code quality or code functionality to diminish ("decay") over time. Effort must be taken to maintain any code or artifacts you add to your repositories, including those which are unused. Consider using [Vulture](https://github.com/jendrikseipp/vulture), [Pylint](https://pylint.pycqa.org/), or [Coverage](https://coverage.readthedocs.io/) to help illuminate sections of your code which may need to be removed.
+
```
[the remark this answers is not in the document]
Comment (d33bs) [reply _posts/2022-12-12-Remove-Unused-Code-to-Avoid-Decay.md:24]: Cheers - thanks, trying to treat the audience's time with care 🙂 .

File: _posts/2022-12-12-Remove-Unused-Code-to-Avoid-Decay.md (hunk 1 above)
Comment (d33bs) [reply _posts/2022-12-12-Remove-Unused-Code-to-Avoid-Decay.md:111]: Thanks, great point, I'll add this change into a forthcoming commit.

File: _posts/2022-12-12-Remove-Unused-Code-to-Avoid-Decay.md (hunk 2 above)
Comment (d33bs) [reply _posts/2022-12-12-Remove-Unused-Code-to-Avoid-Decay.md:119]: Thank you, this makes sense to me and feels like it flows better. I'll add this in.

**Comment by d33bs** [comment]
Thank you @falquaddoomi and @seandavi !

**Pull request merged**